### PR TITLE
StoreKeyConverter in ambry-tools, changed StoreKeyConverter interface slightly

### PR DESCRIPTION
**Pull request by dharju**

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java [hunk 1]
```diff
@@ -27,13 +27,10 @@
...
-   * (Note: TBD if we should return {@code null} if already in format requested but it is useful to diffrentiate
-   * b/w the cases)
+   * desired or no mapping exists, returns the same {@link StoreKey} as the mapping.
    * @param input the {@link StoreKey}s that need to be converted.
    * @return a mapping for each {@link StoreKey} in the new format.
    * @throws Exception that may be thrown when performing the conversion operation
    */
-  public Map<StoreKey, StoreKey> convert(Collection<StoreKey> input) throws Exception;
+  public Map<StoreKey, StoreKey> convert(Collection<? extends StoreKey> input) throws Exception;
 }
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java]: minor: newline required

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java [hunk 2]
```diff
@@ -151,21 +155,23 @@ public static void main(String args[]) throws Exception {
       }
       Time time = SystemTime.getInstance();
       Throttler throttler = new Throttler(config.indexEntriesToProcessPerSec, 1000, true, time);
+      StoreKeyConverterFactory storeKeyConverterFactory = Utils.getObj(serverConfig.serverStoreKeyConverterFactory, properties, null);
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: pass the metric registry? `clusterMap.getMetricRegistry()`

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java [hunk 3]
```diff
@@ -211,20 +217,44 @@ public ConsistencyCheckerTool(ClusterMap clusterMap, StoreKeyFactory storeKeyFac
...
+   * through the StoreKeyConverter in a batch operation
+   * @param replicas An Array of replica directories from which blobIds need to be collected
+   * @param results the results of processing the indexes of the given {@code replicas}.
+   * @return mapping of original keys to converted keys.  If there's no converted equivalent
+   * the value will be the same as the key.
+   * @throws Exception
+   */
+  private Map<StoreKey, StoreKey> createConversionKeyMap(File[] replicas,
+    Map<File, DumpIndexTool.IndexProcessingResults> results) throws Exception {
+    List<StoreKey> storeKeys = new ArrayList<>();
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: Isn't a `Set` better here? The keys will be repeated across stores.

File: ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java [hunk 4]
```diff
@@ -157,10 +158,12 @@ public static void main(String[] args) throws Exception {
     DiskReformatterConfig config = new DiskReformatterConfig(properties);
     StoreConfig storeConfig = new StoreConfig(properties);
     ClusterMapConfig clusterMapConfig = new ClusterMapConfig(properties);
+    ServerConfig serverConfig = new ServerConfig(properties);
     ClusterAgentsFactory clusterAgentsFactory =
         Utils.getObj(clusterMapConfig.clusterMapClusterAgentsFactory, clusterMapConfig, config.hardwareLayoutFilePath,
             config.partitionLayoutFilePath);
     try (ClusterMap clusterMap = clusterAgentsFactory.getClusterMap()) {
+      StoreKeyConverterFactory storeKeyConverterFactory = Utils.getObj(serverConfig.serverStoreKeyConverterFactory, properties, null);
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java]: pass in the metric registry

File: ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java (hunk 1 above)
Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java]: You mean in the comments?

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java (hunk 2 above)
Comment (dharju) [reply ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: StoreKeyConverterFactory doesn't currently use it, but another implementation may.  Fixed.

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java (hunk 3 above)
Comment (dharju) [reply ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: fixed

File: ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java (hunk 4 above)
Comment (dharju) [reply ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java]: fixed

File: ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java (hunk 1 above)
Comment (vgkholla) [reply ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java]: No - the file. Doesn't have a newline at the end

Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/StoreKeyConverter.java]: ah ok. fixed

**Comment by dharju** [comment]
Confirmed working for two replicas

**Pull request merged**